### PR TITLE
[5.1] Convert mod_latestactions to service provider

### PR DESCRIPTION
**Pull request by joomlaweby**
### Summary of Changes

Converts the administrator latest actions module to service provider.

### Testing Instructions

1. Make sure you have administrator module mod_latestactions published
2. You should see module output in administrator area
3. You should not see module output, when logged in to administrator without SuperUser permissions

mod_latestactions

### Actual result BEFORE applying this Pull Request

Module works as expected

### Expected result AFTER applying this Pull Request

Module works same way as before applying pull request

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by fgsw** [comment]
I have tested this item :white_check_mark: successfully on f8b5f0eaefccf74a40980c316d4937ec91e6f376<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42910">issues.joomla.org/tracker/joomla-cms/42910</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on f8b5f0eaefccf74a40980c316d4937ec91e6f376<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42910">issues.joomla.org/tracker/joomla-cms/42910</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42910">issues.joomla.org/tracker/joomla-cms/42910</a>.</sub>

File: administrator/modules/mod_latestactions/src/Helper/LatestActionsHelper.php [hunk 1]
```diff
@@ -69,9 +83,37 @@ public static function getList(&$params)
      * @return  string    The alternate title for the module.
      *
      * @since   3.9.1
+     *
+     * @deprecated __DEPLOY_VERSION__ will be removed in 6.0
```
Comment (bembelimen) [inline administrator/modules/mod_latestactions/src/Helper/LatestActionsHelper.php]:
```suggestion
     * @deprecated __DEPLOY_VERSION__ will be removed in 7.0
```

File: administrator/modules/mod_latestactions/src/Helper/LatestActionsHelper.php [hunk 2]
```diff
@@ -69,9 +83,37 @@ public static function getList(&$params)
...
+     *
+     * @param   Registry  &$params  The module parameters.
+     *
+     * @return  mixed  An array of action logs, or false on error.
+     *
+     * @since   3.9.1
+     *
+     * @throws  \Exception
+     *
+     * @deprecated __DEPLOY_VERSION__ will be removed in 6.0
```
Comment (bembelimen) [inline administrator/modules/mod_latestactions/src/Helper/LatestActionsHelper.php]:
```suggestion
     * @deprecated __DEPLOY_VERSION__ will be removed in 7.0
```

**Comment by bembelimen** [comment]
Thx

**Pull request merged**